src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -42,471 +44,524 @@ public static class SystemNamespaceType
-                .Build(),
+                .Build();
 
-            new FunctionOverloadBuilder("loadFileAsBase64")
+            yield return new FunctionOverloadBuilder("loadFileAsBase64")
                 .WithGenericDescription($"Loads the specified file as base64 string. File loading occurs during compilation, not at runtime. The maximum allowed size is {LanguageConstants.MaxLiteralCharacterLimit / 4 * 3 / 1024} Kb.")
                 .WithRequiredParameter("filePath", LanguageConstants.String, "The path to the file that will be loaded")
                 .WithReturnResultBuilder(LoadContentAsBase64ResultBuilder, LanguageConstants.String)
                 .WithEvaluator(StringLiteralFunctionReturnTypeEvaluator)
                 .WithVariableGenerator(StringLiteralFunctionVariableGenerator)
-                .Build(),
-            new FunctionOverloadBuilder("loadJsonContent")
+                .Build();
+            yield return new FunctionOverloadBuilder("loadJsonContent")
                 .WithGenericDescription($"Loads the specified JSON file as bicep object. File loading occurs during compilation, not at runtime.")
                 .WithRequiredParameter("filePath", LanguageConstants.String, "The path to the file that will be loaded")
                 .WithOptionalParameter("jsonPath", LanguageConstants.String, "JSONPath expression to narrow down the loaded file. If not provided, a root element indicator '$' is used")
                 .WithOptionalParameter("encoding", LanguageConstants.LoadTextContentEncodings, "File encoding. If not provided, UTF-8 will be used.")
                 .WithReturnResultBuilder(LoadJsonContentResultBuilder, LanguageConstants.Any)
                 .WithEvaluator(JsonContentFunctionReturnTypeEvaluator)
                 .WithVariableGenerator(JsonContentFunctionVariableGenerator)
-                .Build(),
+                .Build();
 
-            new FunctionOverloadBuilder("items")
+            yield return new FunctionOverloadBuilder("items")
                 .WithGenericDescription("Returns an array of keys and values for an object. Elements are consistently ordered alphabetically by key.")
                 .WithRequiredParameter("object", LanguageConstants.Object, "The object to return keys and values for")
                 .WithReturnResultBuilder(ItemsResultBuilder, GetItemsReturnType(LanguageConstants.String, LanguageConstants.Any))
-                .Build(),
-        }.ToImmutableArray();
+                .Build();
+
+            if (featureProvider.LambdasEnabled)
+            {
+                yield return new FunctionOverloadBuilder("flatten")
+                    .WithGenericDescription("Flattens an array of arrays into an array containing all of the individual array elements in the original order.")
+                    .WithVariableParameter("array", LanguageConstants.Array, 0, "TBD")
+                    .WithReturnType(LanguageConstants.Array)
+                    .Build();
```

Comment:
It's part of the same release train in ARM - so will be available at the same time lambdas are available. I could probably rename the feature to something like `EnhancedListComprehensionFunctions`?

Reply:
That makes sense to me; no need to rename the feature flag (especially since the plan is to remove it shortly). I just wanted to make sure this wasn't a copy/pasta error